pypika/terms.py

```diff
@@ -455,6 +450,124 @@ def get_sql(self, **kwargs):
+        else:
+            name = self.name
+
+        return '{quote}{val}{quote}'.format(
+            quote=self.quote_char,
+            val=name,
+        )
+
+    def haskey(self, other):
+        other = self._base_prepare(other)
+        return BasicCriterion(PostgresOperators.HAS_KEY, self, other)
+
+    def contains(self, other):
+        other = self._base_prepare(other)
+        return BasicCriterion(PostgresOperators.CONTAINS, self, other)
+
+    def contained_by(self, other):
+        other = self._base_prepare(other)
+        return BasicCriterion(PostgresOperators.CONTAINED_BY, self, other)
+
+    def has_keys(self, other):
+        if isinstance(other, list):
+            other = JSONField(Array(*other), quote_char='')
+        else:
+            raise QueryException('Type `{}` does not support for this operation'.format(type(other)))
+        return BasicCriterion(PostgresOperators.HAS_KEYS, self, other)
+
+    def has_any_keys(self, other):
+        if isinstance(other, list):
+            other = JSONField(Array(*other), quote_char='')
+        else:
+            raise QueryException('Type `{}` does not support for this operation'.format(type(other)))
+        return BasicCriterion(PostgresOperators.HAS_ANY_KEYS, self, other)
+
+    def get_value_by_key(self, other):
+        if isinstance(other, str):
+            other = JSONField(other, quote_char=self.EXPRESSION_QUOTE_CHAR)
+        elif isinstance(other, list) and len(other) == 2:
+            root_key = JSONField(other[0], quote_char=self.EXPRESSION_QUOTE_CHAR)
+            nested_key = JSONField(other[1], quote_char=self.EXPRESSION_QUOTE_CHAR)
+            return NestedCriterion(
+                PostgresOperators.GET_JSON_VALUE, PostgresOperators.GET_TEXT_VALUE,
+                self, root_key, nested_key,
+            )
+        else:
+            raise QueryException('Type `{}` does not support for this operation'.format(type(other)))
+        return BasicCriterion(PostgresOperators.GET_JSON_VALUE, self, other)
+
+    def _base_prepare(self, other):
+        if isinstance(other, str):
+            other = JSONField(other, quote_char=self.EXPRESSION_QUOTE_CHAR)
+        elif isinstance(other, (dict, list)):
+            other = JSONField(json.dumps(other), quote_char=self.EXPRESSION_QUOTE_CHAR)
+        elif isinstance(other, JSONField):
+            other.quote_char = self.EXPRESSION_QUOTE_CHAR
+        else:
+            raise QueryException('Type `{}` does not support'.format(type(other)))
+
+        return other
+
+
+class NestedCriterion(Criterion):
+    def __init__(self, comparator, nested_comparator, left, right, nested, alias=None):
+        super().__init__(alias)
+        self.left = left
+        self.comparator = comparator
+        self.nested_comparator = nested_comparator
+        self.right = right
+        self.nested = nested
+
+    def fields(self):
+        return [*self.right.fields(), *self.left.fields(), *self.nested.fields()]
```

Comment:
yeah, I'll fix it asap.

Reply:
I changed it to `itertools.chain` method. And added `kwargs.setdefault('dialect', self.dialect)` for `QueryBuilder.get_sql` because seems like `get_sql` not always equal `__str__` method if run it without parameters.